File: src/test/java/com/kiwi/mobile/retrograph/StringConverterFactory.java
package com.kiwi.mobile.retrograph;

import java.lang.annotation.*;
import java.lang.reflect.*;

import retrofit2.*;

import okhttp3.*;

final class StringConverterFactory extends Converter.Factory {
  @Override
  public Converter<ResponseBody, ?> responseBodyConverter(
    Type type, Annotation[] annotations, Retrofit retrofit
  ) {
    return (Converter<ResponseBody, String>) ResponseBody::string;
  }

  @Override
  public Converter<?, RequestBody> requestBodyConverter(
    Type type, Annotation[] parameterAnnotations, Annotation[] methodAnnotations,
    Retrofit retrofit
  ) {
    return (Converter<String, RequestBody>) value ->
      RequestBody.create(MediaType.parse("text/plain"), value);
  }
}

File: src/test/java/com/kiwi/mobile/retrograph/converter/StringConverterFactory.kt
package com.kiwi.mobile.retrograph.converter

import okhttp3.*
import okhttp3.MediaType.Companion.toMediaTypeOrNull

import retrofit2.*

import java.lang.reflect.*

class StringConverterFactory: Converter.Factory() {

  override fun responseBodyConverter(
    type: Type,
    annotations: Array<Annotation>,
    retrofit: Retrofit
  ) =
    Converter<ResponseBody, String> { it.string() }

  override fun requestBodyConverter(
    type: Type,
    parameterAnnotations: Array<Annotation>,
    methodAnnotations: Array<Annotation>,
    retrofit: Retrofit
  ) =
    Converter<String, RequestBody> { value ->
      "text/plain".toMediaTypeOrNull()
        ?.let { RequestBody.create(it, value) }
    }
}
